src/ol/renderer/Map.js

```diff
@@ -276,22 +276,26 @@ class MapRenderer extends Disposable {
    * @protected
    */
   scheduleExpireIconCache(frameState) {
-    frameState.postRenderFunctions.push(/** @type {import("../PluggableMap.js").PostRenderFunction} */ (expireIconCache));
+    if (iconImageCache.canExpireCache()) {
+      frameState.postRenderFunctions.push(expireIconCache);
+    }
   }
 
   /**
    * @param {!import("../PluggableMap.js").FrameState} frameState Frame state.
    * @protected
    */
   scheduleRemoveUnusedLayerRenderers(frameState) {
+    const layerStatesMap = getLayerStatesMap(frameState.layerStatesArray);
+    const layerKeys = [];
     for (const layerKey in this.layerRenderers_) {
-      if (!(layerKey in frameState.layerStates)) {
-        frameState.postRenderFunctions.push(
-          /** @type {import("../PluggableMap.js").PostRenderFunction} */ (this.removeUnusedLayerRenderers_.bind(this))
-        );
-        return;
+      if (!(layerKey in layerStatesMap)) {
+        layerKeys.push(layerKey);
       }
     }
+    if (layerKeys.length > 0) {
+      frameState.postRenderFunctions.push(this.removeUnusedLayerRenderers_.bind(this, layerKeys));
```

Comment:
We could just get rid of the `removeUnusedLayerRenderers_` method altogether and push an anonymous function in the loop above.  Something like:
```js
if (!(layerKey in layerStatesMap)) {
  frameState.postRenderFunctions.push(function() {
    this.removeLayerRendererByKey(layerKey).dispose();
  }.bind(this));
}
```

Reply:
yes, that's a very good idea